watertap/tools/oli_api/flash.py

```diff
@@ -0,0 +1,494 @@
+
+        :return result: dictionary containing IDs and output streams for each flash calculation
+        """
+        _logger.info("Running flash calculations")
+        if survey is None:
+            _logger.info("Running single flash calculation")
+            result = {
+                0: oliapi_instance.call(
+                    "POST", flash_method, dbs_file_id, initial_input
+                )
+            }
+
+        else:
+            clones = self.modify_inputs(
+                initial_flash_input=initial_input,
+                survey=survey,
+                flash_method=flash_method,
+            )
+            _logger.info("Running flash survey with {} samples".format(len(clones)))
+            result = {}
+            for k, v in clones.items():
+                _logger.info("Running sample #{} of {}".format(k + 1, len(clones)))
+                result[k] = oliapi_instance.call("POST", flash_method, dbs_file_id, v)
+
+        _logger.info("Completed running flash calculations")
+        if file_name:
+            self.write_output(result, file_name)
+
+        return result
+
+    def modify_inputs(self, initial_flash_input, survey, flash_method):
+        """
+        Iterates over a survey to create modified clones of an initial flash analysis output.
+
+        :param initial_flash_input: flash analysis input to copy
+        :param survey: dictionary containing modifications for each test
+        :param flash_method: string name of flash method to use
+
+        :return clones: dictionary containing modified state variables and survey index
+        """
+
+        clones = {}
+        for i in survey.keys():
+            modified_clone = deepcopy(initial_flash_input)
+            if flash_method == "wateranalysis":
+                for param in modified_clone["params"]["waterAnalysisInputs"]:
+                    if param["name"] in survey[i].keys():
+                        param.update({"value": survey[i][param["name"]]})
+            elif flash_method == "isothermal":
+                for param in survey[i]:
+                    if param in ["Temperature", "Pressure"]:
+                        modified_clone["params"][param.lower()]["value"] = survey[i][
+                            param
+                        ]
+                    elif param in modified_clone["params"]["inflows"]["values"]:
+                        if self.relative_inflows:
+                            modified_clone["params"]["inflows"]["values"][
+                                param
+                            ] += survey[i][param]
+                        else:
+                            modified_clone["params"]["inflows"]["values"][
+                                param
+                            ] = survey[i][param]
+                    else:
+                        raise ValueError(
+                            "Only composition and temperature/pressure surveys are currently supported."
```

Comment:
Any reason why a pH survey isn't supported here? We can do this betting setting reconciliation option on for pH calculation, from what I recall.

Reply:
I figured a pH sweep could be done by creating a survey for H+ or OH- for now. The alternative reconciliation options are in the code but they haven't been tested, so I didn't include advanced functionality for them

Reply:
Again- not something necessary for this PR of course. Rather, we can consider adding this to an issue that tracks all OLI-related enhancements that we want to incorporate in subsequent PRs.

Reply:
That's a great idea